Decode/PerfItemType.cs

```diff
@@ -6,90 +6,151 @@ namespace Microsoft.LinuxTracepoints.Decode
     /// </item>
+    ///
     /// <item>
-    /// The beginning or end of a structure (non-array field, or one element of an array field):
-    /// ElementCount is 1.
-    /// TypeSize is 0.
-    /// Format should be ignored.
-    /// StructFieldCount is significant.
+    /// <b>Complex scalar:</b> <c>IsScalar &amp;&amp; TypeSize == 0</c>
+    /// <br/>
+    /// Non-array field, or one element of an array field.
+    /// Value type is complex (variable-size or struct value).
+    /// <br/>
+    /// <c>ElementCount</c> is always 1.
+    /// <br/>
+    /// If <c>Encoding == Struct</c>, this is the beginning or end of a structure,
+    /// <c>Format</c> should be ignored, and <c>StructFieldCount</c> is significant.
+    /// Otherwise, this is a variable-length value, <c>Format</c> is significant,
+    /// and <c>StructFieldCount</c> should be ignored.
     /// </item>
+    ///
     /// <item>
-    /// The beginning of an array of simple type (non-struct, element's type is fixed-size):
-    /// ElementCount is the number of elements in the array.
-    /// TypeSize is the size of the element's type.
-    /// Format is significant.
-    /// StructFieldCount should be ignored.
+    /// <b>Simple array:</b> <c>!IsScalar &amp;&amp; TypeSize != 0</c>
+    /// <br/>
+    /// Array field (array-begin or array-end item).
+    /// Array element type is simple (fixed-size element).
+    /// <br/>
+    /// <c>ElementCount</c> is the number of elements in the array.
+    /// <br/>
+    /// <c>Format</c> is significant and <c>StructFieldCount</c> should be ignored
+    /// (simple type is never <c>Struct</c>).
     /// </item>
+    ///
     /// <item>
-    /// The end of an array of simple type:
-    /// ElementCount is the number of elements in the array.
-    /// TypeSize is the size of the element's type.
-    /// Format is significant.
-    /// StructFieldCount should be ignored.
-    /// </item>
-    /// <item>
-    /// The beginning or end of an array of complex elements:
-    /// ElementCount is the number of elements in the array.
-    /// TypeSize is 0.
-    /// Either Format or StructFieldCount is significant, depending on whether the Encoding is Struct.
+    /// <b>Complex array:</b> <c>!IsScalar &amp;&amp; TypeSize == 0</c>
+    /// <br/>
+    /// Array field (array-begin or array-end item).
+    /// Array element type is complex (variable-size or struct element).
+    /// <br/>
+    /// <c>ElementCount</c> is the number of elements in the array.
+    /// <br/>
+    /// If <c>Encoding == Struct</c>, this is the beginning or end of an array of structures,
+    /// <c>Format</c> should be ignored, and <c>StructFieldCount</c> is significant.
+    /// Otherwise, this is an array of variable-length values, <c>Format</c> is significant,
+    /// and <c>StructFieldCount</c> should be ignored.
     /// </item>
+    ///
     /// </list>
     /// </summary>
-    public readonly ref struct PerfItemType
+    public readonly ref struct PerfItemMetadata
```

Comment:
Note: Renamed type `PerfItemType` to `PerfItemMetadata`. TODO in a subsequent PR: rename the .cs file accordingly. (Not done now because it makes reviewing the PR impossible.)